src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -170,6 +171,7 @@ public int GetHashCode(object obj)
     }
 
     [Serializable]
+    [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
     internal sealed class CultureAwareComparer : StringComparer
```

Comment:
Wes is your concern the amount of work, the risk for 2.0, or the long term maintenance? As @stephentoub mentioned, it's a goal for 2.0 to support roundtripping. Viktor can work on it, and we still have time, it seems to me.